### DISCUSSION TITLE
WIP-001: GPS Penalty Adjustment

### DISCUSSION THREAD

**Discussion by LeoShiroe** [weatherxm-network/weatherxm-network.github.io#29] [Stage 2: Proposals]
# WIP-001: GPS Penalty Adjustment

## Title

**Adjustment of GPS Location Requirements to Reduce Penalties**

## Abstract

This proposal aims to modify the GPS requirements for WeatherXM weather stations, changing the criteria from daily updates to a three-day window before penalties are applied. This change will reduce unnecessary penalties for occasional missed updates while maintaining data integrity and network reliability.

## Motivation

Currently, WeatherXM WiFi and 4G stations must provide GPS location updates at least daily, and Helium stations at least once every two days, with penalties applied for missed updates. This can lead to undue penalization for minor disruptions, decreasing rewards and potentially discouraging participation. By allowing a three-day window, we can provide additional flexibility to station operators and improve overall network efficiency without compromising the quality of the data.

## Specification

**Current Requirement:** Weather stations must send a GPS location update at least once per day or every two days.

**Proposed Change:** Weather stations will continue to send GPS location updates daily. However, penalties will only be applied if a station fails to send a GPS update for three consecutive days.

## Rationale

- **Flexibility:** A three-day penalty window allows stations to manage minor disruptions or maintenance without immediate penalties, thus improving station uptime and operator satisfaction.
- **Network Reliability:** Maintaining daily updates ensures data remains accurate and reliable. The three-day penalty window balances flexibility with the need for consistent data.
- **Operational Efficiency:** Reducing the frequency of penalties can lower administrative burdens and costs, making the network more efficient and user-friendly.

## Implementation

1. **Policy Update:** Revise the current penalty logic within the Proof of Location (PoL) and rewards distribution algorithms to enforce the three-day window for GPS updates. This involves adjusting the criteria for when penalties are triggered based on consecutive missed updates.
2. **System Adjustment:** Update the PoL and rewards distribution algorithms to track GPS updates over a rolling three-day period. Penalties will only be applied if no GPS updates are received for three consecutive days.
3. **Communication:** Notify all weather station operators of the new policy, providing detailed guidelines and support for understanding and adapting to the change.

## Affected Stakeholders

- **Weather Station Operators:** The primary stakeholders who will benefit from increased flexibility and reduced penalties. They may need to adjust their operational routines to align with the new policy.
- **Data Consumers:** Entities that rely on WeatherXM data, such as businesses and researchers, will benefit from continued data reliability without the disruptions caused by frequent penalties.
- **WeatherXM Network:** Will need to implement and monitor the new policy, ensuring compliance and addressing any issues that arise during the transition.

## Potential Downsides

- **Data Gaps:** Extending the penalty window might lead to slightly longer periods without GPS updates, potentially affecting data granularity and real-time accuracy for certain applications.
- **Implementation Challenges:** Updating the PoL and rewards distribution algorithms to accommodate the new policy may require significant development and testing efforts.
- **Initial Confusion:** Operators and other stakeholders may initially face confusion or challenges in adapting to the new policy, requiring effective communication and support from WeatherXM administration.

## References

- [WeatherXM Documentation](https://docs.weatherxm.com)
- [WeatherXM Proof of Location](https://docs.weatherxm.com/rewards/proof-of-location)

---

By adjusting the penalty criteria, we aim to improve the user experience of the WeatherXM network and remove effects discouraging participation in the network while maintaining the integrity and reliability of the weather data collected. This change is expected to encourage broader participation and reduce unnecessary penalties for occasional missed GPS updates.

**Comment by PavlosTze** [comment]
I agree 100% with that WIP proposal. Given the fact that the strong majority of the users setup a station and never relocate it, I think it's safe to assume that we can have more "relaxed" rules for the required location packets in order to pass successfully our PoL algorithm's requirements. 

This proposal will lead towards less unjustified penalties which can come after external disruptions. We should always keep in mind that WeatherXM is not responsible nor can do anything to fix them or predict them. 

So the only "best effort" fix we can do is this proposal which I agree that aligns with the interests of every party.

**Comment by cjohlandt** [comment]
I also agree with this WIP proposal.  Given the challenges with clean GPS updates, it is unfair to penalize station owners who have not relocated their stations just because a few packets don't make it through the network.  I would like to request that the WXM team implement a notification system that would alert station owners if the daily GPS location update is not received so that they can at least check to see if anything needs to be fixed on their end and reduce the risk of lost rewards.

**Comment by Skelly85** [comment]


**Reply by LeoShiroe** [reply]
Hey Steve! Thank you for your contribution!

To be clear, this WIP does not change anything in data frequency or GPS frequency; It's a simple change in the rewards mechanism only, the stations will continue to operate as they have before, sending location data daily.

Regarding the statistics, we don't have exact numbers on it, but the overwhelming majority of GPS cases are temporary loss of GPS for a couple days with no reason attributed to it; Likely caused by nearby interference, harsh weather, nearby Hotspot issues. 

Usually, when a station does not get GPS because it is poorly deployed, the way it behaves isn't having a 2 to 3 day gap and then go back to normally sending a GPS location; It simply does not send GPS coordinates until it is properly deployed, or maybe once in a while it is able to get a single GPS packet.

Regarding masking Helium router issues, I believe it station owners shouldn't be penalised at 100% loss in rewards in order to uncover these issues. They must be uncovered, and the best indicator is data packet instability. We don't need to penalise stations on a GPS basis to uncover these issues.
 
Regarding having higher rewards for some stations, that'd go against the whole point of this WIP; Having more stable rewards helps the growth of the network. And once again, there will be no change in reporting frequency.

**Comment by Gooqii** [comment]
I support this WIP as I have had at least 2 zero reward events. The last was 2 days of zero rewards with the issue being "No location data". 

Is it an option to retrospectively check GPS data. Lets say over a 2 day period no GPS data is received and on the 3rd GPS is back with the exact same location as was "known" before. If this is the case then Rewards for the lost 2 days can be restored. I am wondering that if on a Helium Station, if the packet that is sent to Helium is lost, and this is the packet containing GPS data then it is possible that GPS PoL data suffers. (Understand that I do not know the technical details of how the GPS is sent so I might have this view wrong)

All in all this WIP seems it will improve stability of PoL. (@LeonB on discord)

**Reply by vrypan** [reply]
(I removed my previous comment, I misread what you wrote, and it was in the wrong context.)

**Reply by Gooqii** [reply]
All good, I have always said, texting can be difficult to understand.

**Comment by phidauex** [comment]
This WIP seems to be pretty well fleshed out, and I'm not seeing any real dissent. What would be the next steps to advancing it? Does the proposal itself need more detail such as pseudocode for the algorithm?

**Comment by eeVoskos** [comment]
[Submitted to GA](https://weatherxm.network/docs/governance#phase-2-approval-by-the-ga)

**Reply by leafarhz** [reply]
Thank you this is great, I have been loosing rewards in all my Wifi Antenas!

**Comment by AndrewAnduWXM** [comment]
This is a very good idea that needs to be put into practice as soon as possible!

**Comment by eeVoskos** [comment]
This WIP was initially accepted by the GA. The voting was announced and will begin on Friday, April 4, 2025.
https://discord.com/channels/813503633278631946/910124655728295936/1356304152250023956

**Comment by eeVoskos** [comment]
WIP-001 voting started:
https://snapshot.box/#/s:weatherxm.eth/proposal/0xff24e822f42a677c5ce1b8cc00148b353af58debb72cc8fe5872fd43d4a13f75
